src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -252,6 +252,38 @@ internal void SwitchToChain(ChainedHeaderTree cht, ChainedHeader chainTip, Chain
                     cht.ConsensusTipChanged(currentConsumedCh);
                 }
             }
+
+            internal List<BlockHeader> GetListOfAllUniqueBlockHeaders(ChainedHeaderTree cht)
```

Comment:
This was removed from 1608 on purpose. 
If you want to keep the method please replace body with: 

```
internal List<BlockHeader> GetListOfAllUniqueBlockHeaders(ChainedHeaderTree cht)
{
    var allBlockHeaders = new List<BlockHeader>();

    ChainedHeader genesis = cht.GetChainedHeaderBlock(this.Network.GenesisHash).ChainedHeader;

    var headersToProcess = new Stack<ChainedHeader>();
    headersToProcess.Push(genesis);

    while (headersToProcess.Count > 0)
    {
        ChainedHeader current = headersToProcess.Pop();
        allBlockHeaders.Add(current.Header);

        foreach (ChainedHeader next in current.Next)
            headersToProcess.Push(next);
    }

    return allBlockHeaders;
}
```

Reply:
In general we don't use methods declared in a body of another method and trying to avoid recursion

Reply:
okay, didn't know about it.